extensions/wikia/PortableInfobox/services/PortableInfoboxRenderService.class.php

```diff
@@ -230,4 +234,60 @@ private function isEuropaTheme() {
 
 		return !empty( $wgEnablePortableInfoboxEuropaTheme );
 	}
+
+	private function createSmartGroups( $groupData, $rowCapacity ) {
+		$result = [ ];
+		$rowSpan = 0;
+		$rowItems = [ ];
+
+		foreach ( $groupData as $item ) {
+			$data = $item['data'];
+
+			if ( $item['type'] === 'data' && ( !isset( $data['layout'] ) || $data['layout'] !== 'default' ) ) {
```

Comment:
what do you think about moving such complex conditions to separate method? or at least the part after && operator